### DISCUSSION TITLE
Microsoft.WindowsAppSDK and GPLv3 License

### DISCUSSION THREAD

**Discussion by JohnCido** [microsoft/WindowsAppSDK#3511] [Q&A] [answered]
I see that Microsoft.WindowsAppSDK NuGet package has [this license](https://www.nuget.org/packages/Microsoft.WindowsAppSDK/1.2.230217.4/license).

Is it OK that I license my repo under GPLv3? Does this license from Microsoft prevent me from doing so?

**Comment by JohnCido** [comment]
Or it's OK since this repo uses MIT license?

**Comment by rocksdanister** [comment, marked as answer]
https://www.gnu.org/licenses/gpl-faq.en.html#SystemLibraryException

**Reply by fredemmott** [reply]
That seems extremely unlikely to apply:

>  However, as a special exception, the source code distributed need not include anything that is normally distributed (in either source or binary form) with the major components (compiler, kernel, and so on) of the operating system on which the executable runs, unless that component itself accompanies the executable.

... or GPL v3:

> The “System Libraries” of an executable work include anything, other than the work as a whole, that (a) is included in the normal form of packaging a Major Component, but which is not part of that Major Component, and (b) serves only to enable use of the work with that Major Component, or to implement a Standard Interface for which an implementation is available to the public in source code form. A “Major Component”, in this context, means a major essential component (kernel, window system, and so on) of the specific operating system (if any) on which the executable work runs, or a compiler used to produce the work, or an object code interpreter used to run it.

WinUI3 is a just a UI library like any other, often distributed with the app, nearly never with the OS.

---

That said, the Window App SDK repository is now MIT licensed and appears to be buildable - but WinUI3 is not, and does not plan to be. 

WinUI3's source is here: https://github.com/microsoft/microsoft-ui-xaml/

While it is MIT-licensed, it is not buildable, which means it is GPL-incompatible, as it also clearly does not fit the system library exceptions.

From their community call followup at https://github.com/microsoft/microsoft-ui-xaml/discussions/8980 :

> Q: ​Will we be able to build from the source or only view due to private dependencies?
> A: To be clear, we are doing shared source, not open source, as there are underlying interwoven components linking WinUI 3 to the system that we can't share. We will not support building, deploying, or running the source code and will not accept PRs for it. We will support viewing, searching, and debugging of the source code when it is available.

**Reply by sylveon** [reply]
You can add an exception under the section 7 of the GPLv3 license to allow linking the program against the Windows App SDK (which WinUI is part of) and convey the resulting work. See [this](https://www.gnu.org/licenses/gpl-faq.html#GPLIncompatibleLibs).

> Additional permission under GNU GPL version 3 section 7
> 
> If you modify this Program, or any covered work, by linking or combining it with the Windows App SDK (or a modified version of that library), containing parts covered by the terms of the Microsoft Software License, the licensors of this Program grant you additional permission to convey the resulting work.

However, I would imagine if you don't ship the Windows App SDK yourself (e.g. not a self contained app), you probably don't have to do that, as the Windows App SDK is not part of your combined work, it is distributed separately (but it can't hurt to include this exception nonetheless).

**Reply by fredemmott** [reply]
I've had a similar exception for some time - https://github.com/OpenKneeboard/OpenKneeboard/blob/master/LICENSE (originally assuming that it was temporary and MS would deliver on their earlier statements that WinUI3 would actually be open source, not shared source) - but this still means I am unable to use any GPL-licensed libraries, unless they in turn have a similar exception.

Even if I distribute it myself, it seems extremely unlikely that "it can be installed automatically  as a re-usable library from the MS store if you install my MSIX" meets either the GPL v2 or v3's bar for 'system component'
